content/articles/2017-08-purrr-0.2.3.Rmd

```diff
@@ -0,0 +1,176 @@
+This means it is now easier to map functions over vector-like S3
+objects. The reason is twofold:
+
+* `map()` and friends apply functions on the elements of the
+  underlying data structure but that is not always appropriate. An
+  object sometimes contains metadata. A user of such a class probably
+  wants to map over the *elements* of the data of interest rather than
+  over the metadata fields.
+
+* `map()` is type-stable and always returns a list. If you're mapping
+  over an object, chances are you want a similar object back.
+
+`modify()` and its variants solve both these problems via S3 dispatch.
+
+As a developer there are two ways to make your class compatible with
+`modify()` . The easiest is to implement a method for the
+subset-assign operator `[<-` which should be sufficient in most
+cases. Alternatively you can implement methods for the `modify`
+functions themselves as they are all generics.
+
+
+### pluck()
+
+The plucking mechanism used for indexing into data structures with
+`map()` has been extracted into the function `pluck()`. Plucking is
+often more readable to extract an element buried in a deep data
+structure. Compare this syntax-heavy extraction which reads
+non-linearly:
+
+```{r, eval = FALSE}
+accessor(x[[1]])$foo
+```
+
+to the equivalent pluck:
+
+```{r, eval = FALSE}
+x %>% pluck(1, accessor, "foo")
+```
+
+The new function `attr_getter()` generates attribute accessors that
+can be used in `pluck()`:
+
+```{r}
+dfs <- list(iris, mtcars)
+dfs %>% pluck(2, attr_getter("row.names"))
+```
+
+or in mapped indexing:
+
+```{r}
+x <- list(
+  list(vec = letters, df = mtcars),
+  list(vec = LETTERS, df = iris)
+)
+x %>% map(list("df", attr_getter("row.names")))
+```
+
+
+### Useful tweaks and features
+
+* `set_names()` (now reexported from rlang) can now take a function to
+  tranform the names programmatically, and you can supply names in
+  `...` to reduce typing even more more.
+
+* You can now refer to arguments by position in lambda-formulas. The
+  `..1` symbol refers to the first argument, `..2` to the second and
+  so on. This makes it easier to use functions like `pmap()` with the
+  formula shortcut.
+
+* All typed map functions now treat `NULL` the same way as an empty
+  vector: they return an empty vector if any input is empty.
+
+* `auto_browse(f)` returns a new function that automatically calls
+  `browser()` if `f` throws an error.
+
+* `vec_depth()` computes the depth (i.e. the number of levels of
+  indexing) or a vector.
+
+* `reduce2()` and `reduce2_right()` make it possible to reduce with a
+  3 argument function where the first argument is the accumulated
+  value, the second argument is `.x`, and the third argument is `.y`.
+
+* `list_modify()` extends `stats::modifyList()` to replace by position
+  if the list is not named. `list_merge()` operates similarly to
+  `list_modify()` but combines instead of replacing.
+
+* `modify_depth()` (previously `at_depth()`) gains new `.ragged`
```

Comment:
I think you can remove this too.

Generally, if it's not important enough to include a small example for, it's probably not worth including here.